### DISCUSSION TITLE
Rendering multiple GEOJSON and saving the draw

### DISCUSSION THREAD

**Discussion by cubecleveland** [geoman-io/leaflet-geoman#1063] [Questions & Help] [answered]
Hello, we have a map with an outside select control that switched between Geojson layers on control select.
When selecting an option the entire state changes and a new Geojson layer is being rendered.

Each layer also applies a drawing control. the idea is that every layer can be edited and then saved to the disk.

All is working perfetcly. the issue is that when we edit a certain layer and for exmapl add a shape. that shape is being added to the main map so switching to another layer keep the newly added shape in place.
Can we make the plugin add the shape to a specific layer rather then creating a new layer ?

**Comment by Falke-Design** [comment]
I think you search for the `layerGroup` [option](https://github.com/geoman-io/leaflet-geoman#options):
```
map.pm.setGlobalOptions({ layerGroup: YOUR_LAYER_GROUP });
```

**Comment by cubecleveland** [comment]
Thank you so much for your answer.

I tries it and it seems like this solution doesn't work with the flow.

The geojson that is being rendered per selection alwasy changes depending on the state so its basially the same layer,

I was thiking about destorying the table on save and renrender it.... but that seems wrong... any thing else we can do ?

**Reply by Falke-Design** [reply]
I don't really understand what you mean but have you tried to call `setGlobalOptions` int he same function as the geojson is changed and made visible? How does the user change the geojson?

**Reply by cubecleveland** [reply]
The GeoJosn is being rendered via the React leaflet library with the GeoJson compoenent and the data inside the geojson is being changed upon the state of the select.  <GeoJosn data={state.jeojsondata}/>
As i understand it the referndce to the layer remains the same.....

**Reply by Falke-Design** [reply]
Can you upload somewhere your code? Maybe you can create a sandbox on stackblitz or so

**Comment by cubecleveland** [comment, marked as answer]
This worked out exactly as you described. ended up "call setGlobalOptions int he same function as the geojson is changed" with a unique ref.
Thanks so much.
